docs: unify installation in one document (#2793)
SumoLogic/sumologic-kubernetes-collection: docs/installation.md

  - [Collecting Kubernetes events](#collecting-kubernetes-events)
- [Viewing Data In Sumo Logic](#viewing-data-in-sumo-logic)
- [Troubleshooting Installation](#troubleshooting-installation)
  - [General troubleshooting](#general-troubleshooting)
  - [Error: timed out waiting for the condition](#error-timed-out-waiting-for-the-condition)
    - [Error: collector with name 'sumologic' does not exist](#error-collector-with-name-sumologic-does-not-exist)
  - [OpenTelemetry Collector Pods Stuck in CreateContainerConfigError](#opentelemetry-collector-pods-stuck-in-createcontainerconfigerror)
  - [Prometheus Troubleshooting](#prometheus-troubleshooting)
- [Upgrading Sumo Logic Collection](#upgrading-sumo-logic-collection)
- [Uninstalling Sumo Logic Collection](#uninstalling-sumo-logic-collection)
  - [Post installation cleanup](#post-installation-cleanup)
  - [Removing the kubelet Service](#removing-the-kubelet-service)
<!-- /TOC -->

Our Helm chart deploys Kubernetes resources for collecting Kubernetes logs, metrics, and events;
enriching them with deployment, pod, and service level metadata; and sends them to Sumo Logic.

## Requirements

### Helm

> Helm is required but you can use it for template generation if you don't want to use it to manage your installation.

Reviewer: ```suggestion
Helm is required, but you can use it for template generation only, if you don't want to use it to manage your installation.
```

This is a one time setup per Sumo Logic account.

## Add repository

> Before installing the chart, you need to add `sumologic` Helm repo:

Reviewer: ```suggestion
Before installing the chart, you need to add the `sumologic` Helm repo:
```

```bash
helm repo add sumologic https://sumologic.github.io/sumologic-kubernetes-collection
helm repo update
```

## Prepare minimal configuration

> Next you can prepare `user-values.yaml` with configuration.

Reviewer: ```suggestion
Next you can prepare a `user-values.yaml` file with your configuration.
```

### Required parameters

The Helm chart installation requires two parameter overrides:

- `sumologic.accessId` - Sumo [Access ID](https://help.sumologic.com/docs/manage/security/access-keys/).
- `sumologic.accessKey` - Sumo [Access key](https://help.sumologic.com/docs/manage/security/access-keys/).

### Proxy

If you are installing the collection in a cluster that requires proxying outbound requests,
please see the following [additional properties](installing-behind-proxy.md) you will need to set.

### Installing with existing Prometheus

> If there Prometheus is already in your cluster,

Reviewer: ```suggestion
If you already have Prometheus installed in your cluster, and would like to use it instead of the Prometheus instance provided by the Chart,
```

please see [Prometheus document](/docs/prometheus.md#prometheus)

> ### Installing in Openshift platform

Reviewer 2: ```suggestion
### Installing in OpenShift platform
```

      changeowner:
        securityContext:
          privileged: true
tailing-sidecar-operator:
  scc:
    create: true
```

__Notice:__ Prometheus Operator is deployed by default on OpenShift platform,
you may either limit scope for Prometheus Operator installed with Sumo Logic Kubernetes Collection using
`kube-prometheus-stack.prometheusOperator.namespaces.additional` parameter in `user-values.yaml` or
exclude namespaces for Prometheus Operator installed with Sumo Logic Kubernetes Collection
using `kube-prometheus-stack.prometheusOperator.denyNamespaces` in `user-values.yaml`.
For details see [Prometheus document](/docs/prometheus.md#prometheus-operator-in-the-cluster)

> ### Setting cluster name

Reviewer 2: Can we move the `Setting cluster name` section up, just below the `Required parameters` section?

The following parameter is optional, but we recommend setting it.

- `sumologic.clusterName` - An identifier for your Kubernetes cluster.
  This is the name you will see for the cluster in Sumo Logic. Default is `kubernetes`.
  Whitespaces in the cluster name will be replaced with dashes.

### Examples

#### Minimal configuration

An example file with the minimum configuration is provided below.

```yaml
sumologic:
  accessId: ${SUMO_ACCESS_ID}
  accessKey: ${SUMO_ACCESS_KEY}
  clusterName: ${MY_CLUSTER_NAME}
```

> #### Openshift with limiting the scope of the interaction of our Prometheus Operator

Reviewer 2: ```suggestion
#### OpenShift with limiting the scope of the interaction of our Prometheus Operator
```

      changeowner:
        securityContext:
          privileged: true
tailing-sidecar-operator:
  scc:
    create: true
```

> #### Openshift using existing Prometheus Operator which is by default available in `openshift-monitoring` namespace

Reviewer 2: ```suggestion
#### OpenShift using existing Prometheus Operator which is by default available in `openshift-monitoring` namespace
```

> An example configuration for Openshift which uses existing Prometheus Operator:

Reviewer 2: ```suggestion
An example configuration for OpenShift which uses existing Prometheus Operator:
```

before applying the configuration.

## Install chart

Now you can install our chart.
The following command will install the Sumo Logic chart with the release name `my-release`
in the `${NAMESPACE}` namespace.

```bash
helm upgrade \
  --install \
  -n `${NAMESPACE}` \
  --create-namespace \
  -f user-values.yaml \
  my-release \
  sumologic/sumologic
```

> __Note__: If the release exists, it will be upgraded, otherwise it will be installed.
>
> __Note__: If the namespace doesn't exists, it will be created.

## Customizing Installation

All default properties for the Helm chart can be found in our [documentation](/deploy/helm/sumologic/README.md).
We recommend creating a new `user-values.yaml` for each Kubernetes cluster you wish
to install collection on and __setting only the properties you wish to override__.
Once you have customized you should use the following commands to install or upgrade.

```bash
helm upgrade \
  --install \
  -n `${NAMESPACE}` \
  --create-namespace \
  -f user-values.yaml \
  my-release \
  sumologic/sumologic
```

> We documented some often customizations:

Reviewer 2: ```suggestion
We documented some common customizations:
```
